soc/mediatek/mt8xxx/irq.c

```diff
@@ -35,6 +35,12 @@ static const struct device *irq_dev(unsigned int *irq_inout)
 	}
 	__ASSERT_NO_MSG(lvl1 == 2);
 	return DEVICE_DT_GET(DT_INST(1, mediatek_adsp_intc));
+#elif defined(CONFIG_SOC_SERIES_MT8365)
+	/* Controller 0 is on Xtensa vector 1 */
+	if ((*irq_inout & 0xff) == 1) {
+		*irq_inout = (*irq_inout >> 8) - 1;
+	}
+	return DEVICE_DT_GET(DT_INST(0, mediatek_adsp_intc));
```

Comment:
Not a fault of this PR, but the hackery in this file (which really is a wart in Zephyr, we lack a generic IRQ routine generator even though DTS gives us all the information needed to generate it) is getting a little ugly.  Probably time to start thinking about a rework.